### PR TITLE
Fix MVTec format detect function

### PR DESCRIPTION
**Pull request by vinnamkim**
### Summary
 - Ticket no. 105561

### How to test
I changed the existing tests after fixing detectability for MVTec.

### Checklist

- [x] I submit my changes into the `develop` branch
- [x] I have added description of my changes into [CHANGELOG](https://github.com/openvinotoolkit/datumaro/blob/develop/CHANGELOG.md)
- [ ] I have updated the [documentation](
  https://github.com/openvinotoolkit/datumaro/tree/develop/docs) accordingly
- [ ] I have added tests to cover my changes
- [x] I have [linked related issues](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/openvinotoolkit/datumaro/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
- [x] I have updated the license header for each file (see an example below)

```python
# Copyright (C) 2023 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

### CODE REVIEW

File: datumaro/plugins/data_formats/mvtec/importer.py [hunk 1]
```diff
@@ -40,12 +45,27 @@ class MvtecClassificationImporter(MvtecImporter):
...
+
 
 class MvtecSegmentationImporter(MvtecImporter):
     _TASK = MvtecTask.segmentation
     _TASKS = {_TASK: MvtecImporter._TASKS[_TASK]}
+
+    @classmethod
+    def detect(cls, context: FormatDetectionContext) -> FormatDetectionConfidence:
+        super().detect(context)
+        return FormatDetectionConfidence.MEDIUM
```
Comment (wonjuleee) [inline datumaro/plugins/data_formats/mvtec/importer.py:71]: Basically, it looks good to me. But if we think about public MVTec, we only have the masks. We then internally convert this to bbox or label by losing information. So how about setting the confidence priority between them in the similar vein, e.g., segmentation could be more preferable than others?

Comment (vinnamkim) [reply datumaro/plugins/data_formats/mvtec/importer.py:71]: Do you mean setting `mvtec_segmentation` to `FormatDetectionConfidence.MEDIUM`, `mvtec_detection` to `FormatDetectionConfidence.LOW`, and `mvtec_classification` to `FormatDetectionConfidence.LOW`?

Comment (wonjuleee) [reply datumaro/plugins/data_formats/mvtec/importer.py:71]: No, I thought about introducing HIGH confidence. Don't you have any plan to introduce this?

Comment (vinnamkim) [reply datumaro/plugins/data_formats/mvtec/importer.py:71]: Other dataset formats have `MEDIUM` at most. So, I think that it is better to make `Segmentation` get `MEDIUM` and the others to `LOW`.

**Review by wonjuleee — APPROVED** [review]

**Pull request merged**